### PR TITLE
Minor Tweaks to PolygonalRegion

### PR DESCRIPTION
**Pull request by Eric-Vin**
### Description
Minor tweaks to PolygonalRegion's `__eq__` and `hash` functions.

### Issue Link
N/A

### Checklist
- [X] I have tested the changes locally via `pytest` and/or other means
- [X] I have added or updated relevant documentation
- [X] I have autoformatted the code with black and isort
- [X] I have added test cases (if applicable)

### Additional Notes
N/A

### CODE REVIEW

File: src/scenic/core/regions.py [hunk 1]
```diff
@@ -3049,7 +3049,14 @@ def __eq__(self, other):
 
     @cached
     def __hash__(self):
-        return hash((self.polygons, self.orientation, self.z))
+        return hash(
+            (
+                tuple(self._points) if self._points else None,
```
Comment (dfremont) [inline src/scenic/core/regions.py:3054]: This seems like an odd place to normalize the type. We don't want someone to pass a list of random points to `PolygonalRegion`, mutate that list, and have the mutation end up affecting the resulting sampled region. Probably better to convert to `tuple` in the initializer and use plain `self._points` here.

**Review by dfremont — CHANGES_REQUESTED** [review]
Looks good, just a minor comment above.

File: src/scenic/core/regions.py (hunk 1 above)
Comment (Eric-Vin) [reply src/scenic/core/regions.py:3054]: Good call, I'll make the change.

**Review by dfremont — APPROVED** [review]
Looks great!

**Pull request merged**